### ISSUE TITLE
Uploaded GIFs play despite reduced motion preference

### ISSUE THREAD

**Issue by openmindculture** [openmindculture/kleiderordnung-berlin#33]
Conversation and ideas:

- Piccia Neri: pause, stop, hide (WCAG guideline)

- Manuel Matuzovic geht das wohl mit <img srcset media query
https://css-tricks.com/gifs-and-prefers-reduced-motion/

- Wo aber soll das alternative Bild herkommen?

- [ ] Eventuell ein leeres Platzhalterbild als Fallback coden.

- [x] Zusätzlich einen CSS Class Name adden um ggf. darüber zu stylen oder auszublenden.

- [ ] Gibt es dazu ein bestehendes WordPress-Plugin, dass sich zudem nicht mit TinyPNG stören darf?

```php
function wpdocs_addTitleToThumbnail( $html ) {
    $id = get_post_thumbnail_id();
    $alt_text = get_post_meta( $id, '_wp_attachment_image_alt', true );

    $html = str_replace( 'alt=', 'title="' . esc_attr( $alt_text ) . '" alt=', $html );

    return $html;
}
add_filter( 'post_thumbnail_html', '
```

https://developer.wordpress.org/reference/hooks/post_thumbnail_html/

**Comment by openmindculture** [comment]
- TODO: This should better be solved using an existing well-tested popular WordPress plugin that does not conflict with TinyPNG.
- TODO: `.gif` does not equal animation. Apart from unlikely false positives caused by static GIF images, false negatives would not detect animated images in formats like `webp` or `avif`. In theory we'd better inspect file contents (see linked StackOverflow discussions) 
- and don't care about speed and CPU as we will cache the output anyway?
- TODO ... but if we go deeper we'd better explicitly consider what tinyPNG possibly already does and hook into their image sourceset code instead of the WordPress core `post_thumbnail_html` filter ...
- ... which might be more reliable anyway?

https://stackoverflow.com/questions/280658/can-i-detect-animated-gifs-using-php-and-gd

https://stackoverflow.com/questions/1412529/how-do-i-programmatically-check-whether-a-gif-image-is-animated

https://stackoverflow.com/questions/45190469/how-to-identify-whether-webp-image-is-static-or-animated

https://stackoverflow.com/questions/74298761/detect-if-avif-image-is-animated-using-javascript

https://stackoverflow.com/questions/1326890/is-it-possible-to-detect-animated-gif-images-client-side

**Comment by openmindculture** [comment]
Very simple workaround to start with (match `'.gif"'` and add a class name `news__post__image--has-gif` to the container so that we can hide the image in case of preferred reduced motion.

Follow-up: asking the TinyPNG plugin support for best practice / missing options / feature update:

https://wordpress.org/support/topic/add-static-fallback-for-animated-images-if-user-prefers-reduced-motion/

**Comment by openmindculture** [comment]
Workaround is good enough to comply to a11y and don't disturb dizzy people.

Reopen when someone suggests a better alternative.
